src/main/java/seedu/address/model/particpant/Participant.java

```diff
@@ -0,0 +1,213 @@
+package seedu.address.model.particpant;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+
+
+
+/**
+ * Represents a Participant in an event
+ * Guarantees: name, phone, email, address, tags are present and not null, field values are validated, immutable.
+ * Non-guarantees: birthDate - to be added with builder chaining method
+ */
+public class Participant extends Person {
+
+    private BirthDate birthDate = BirthDate.notSpecified();
+    private Set<Note> notes = new HashSet<>();
+    private ArrayList<Person> nextOfKins = new ArrayList<>();
+
+
+    /**
+     * Every field must be present and not null.
+     *
+     * @param name    Name object of the person
+     * @param phone   Phone object of the person
+     * @param email   Email object of the person
+     * @param address Address object of the person
+     * @param tags    tags of the person
+     */
+    public Participant(Name name, Phone phone, Email email, Address address, Set<Tag> tags) {
+        super(name, phone, email, address, tags);
+    }
+
+    /**
+     * Builder chaining method for retuning participant object with certain birthdate.
+     * This is to allow the use of Person template along with Person but adding extension fields.
+     * i.e. new Participant(name, phone, email, address, tags).withBirthDate(year, month, dayOfMonth)
+     *
+     * @param year       this year or before
+     * @param month      month of the year specified
+     * @param dayOfMonth day of the month specified
+     * @return another Participant with same particulars object but with birth date
+     */
+    public Participant withBirthDate(int year, int month, int dayOfMonth) {
+        Participant participantToReturn = new Participant(getName(), getPhone(), getEmail(), getAddress(), getTags());
+        participantToReturn.nextOfKins = nextOfKins;
+        participantToReturn.notes = notes;
+        // set birth Date
+        participantToReturn.birthDate = BirthDate.of(year, month, dayOfMonth);
+
+        return participantToReturn;
+    }
+
+    /**
+     * Builder chaining method for retuning participant object with certain birthdate.
+     * i.e. new Participant(name, phone, email, address, tags).withBirthDate(year, month, dayOfMonth).withNotes(notes)
+     * This chaining method is not required unless the participants were to be set with certain notes set
+     *
+     * @param notes set of Note object
+     * @return this Participant but with new notes
+     */
+    public Participant withNotes(Set<Note> notes) {
```

Comment:
Thinking of migrating withXXX logic to ParticipantBuilder to be used in testing instead